### PR TITLE
[#8933] Instructor: Edit rubric question: adding reorder options using up and down arrows for rows

### LINKED ISSUE #8933: Instructor: Edit rubric question: reorder options using drag and drop
**Current**
The user reorders columns using left and right arrow buttons.

image

**Suggested**
Use drag and drop to move the columns around. The issue can be extended to include other related enhancements to rubric question.

### PR DESCRIPTION
**Pull request by xyliew25**
Fixes #8933

**Outline of Solution**
Now, the rubric question edit form looks like this:

https://user-images.githubusercontent.com/75534016/170855307-b5e67673-2786-4f75-9f54-e2299370aaba.mp4

Note the weird animation when dragging across columns. A hack was implemented based on [this StackOverflow post](https://stackoverflow.com/questions/63493988/angular-drag-and-drop-intersection-e-g-rows-and-columns-in-a-table) as I believe dragging and dropping across rows and columns at the same time has yet to be a feature of Angular Material's DragDropModule, as raised by the community [here](https://github.com/angular/components/issues/22287).

### CODE REVIEW

**Comment by xyliew25** [comment]
Or is this better?

https://user-images.githubusercontent.com/75534016/170873498-65218056-303a-488a-a3eb-7f0b363df0a6.mp4

**Comment by damithc** [comment]
Columns already can be reordered using arrow buttons. Shall we add the same ability for rows? I'm not sure if drag-and-drop is significantly better and worth the effort. One pro with arrow buttons option is we don't have to deal with animation issues (which may differ across browsers too). What do you guys think?

**Comment by xyliew25** [comment]
@damithc I think with the current state of the DragDropModule, perhaps it is better to stick with our current implementation of arrow buttons. We can re-consider the drag and drop feature in the future when it is supported. Although it is somewhat easier to reorder rows and columns with the drag and drop feature, the poor animation may not increase the overall UX anyway. 

Also, I think it's worth adding the arrow buttons for rows. Other views?

**Comment by xyliew25** [comment]
Update:

https://user-images.githubusercontent.com/75534016/170947390-8fc051be-23d2-4553-8356-626d6de0e317.mp4

**Comment by xyliew25** [comment]
Hi @damithc @wkurniawan07 @samuelfangjw and all, would appreciate it if someone could review this PR.

**Review by damithc — APPROVED** [review]
I look at the UI/UX only, and I'm fine with the current version 👍

File: src/web/app/components/question-types/question-edit-details-form/rubric-question-edit-details-form.component.html [hunk 1]
```diff
@@ -22,10 +22,16 @@
...
-            <span id="btn-delete-subquestion" class="input-group-text font-weight-bold cursor-pointer"
-                  (mouseover)="rowToHighlight = i" (mouseleave)="rowToHighlight = -1"
-                  (click)="deleteSubQuestion(i)"><i class="fas fa-times"></i></span>
+          <div class="btn-group-vertical">
+            <button type="button" class="btn btn-light" [disabled]="i === 0 || !isEditable" ngbTooltip="Move row up"
+                    (click)="moveRow(i, i - 1)"><i class="fas fa-arrow-up"></i></button>
+            <button id="btn-delete-subquestion" type="button" class="btn btn-light" (click)="deleteSubQuestion(i)"
+                    (mouseover)="rowToHighlight = i" (mouseleave)="rowToHighlight = -1"
+                    [disabled]="model.rubricSubQuestions.length === 1 || !isEditable"
+            ><i class="fas fa-times"></i></button>
```
Comment (ziqing26) [inline src/web/app/components/question-types/question-edit-details-form/rubric-question-edit-details-form.component.html:31]:
```suggestion
                    [disabled]="model.rubricSubQuestions.length === 1 || !isEditable">
             <i class="fas fa-times"></i>
           </button>
```

**Review by ziqing26 — APPROVED** [review]
A bit of style nitpicks, otherwise it's LGTM!

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**